plugins/npm/src/tasks/npm-publish.ts

```diff
@@ -0,0 +1,82 @@
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { Task } from '@dotcom-tool-kit/types'
+import pacote from 'pacote'
+import { readState } from '@dotcom-tool-kit/state'
+import pack from 'libnpmpack'
+import { publish } from 'libnpmpublish'
+import { styles } from '@dotcom-tool-kit/logger'
+import tar from 'tar'
+import { PassThrough as PassThroughStream } from 'stream';
+
+export const semVerRegex = /^v\d+\.\d+\.\d+(-.+)?/
+export const prereleaseRegex = /^v\d+\.\d+\.\d+(?:-\w+\.\d+)$/
+export const releaseRegex = /^v\d+\.\d+\.\d+$/
+
+type TagType = "prerelease" | "latest"
+
+export default class NpmPublish extends Task {
+  static description = ''
+  
+  getNpmTag(tag: string): TagType {
+    if(!tag) {
+        throw new ToolKitError('CIRCLE_TAG environment variable not found. Make sure you are running this on a release version!')
+    }
+    if(prereleaseRegex.test(tag)) {
+      return 'prerelease'
+    }
+    if(releaseRegex.test(tag)) { 
+      return 'latest'
```

Comment:
**thought**: I never realised we set these tags ourselves when publishing – cool!